engine/src/main/scala/cromwell/engine/workflow/WorkflowStoreActor.scala

```diff
@@ -76,37 +76,35 @@ abstract class WorkflowStore {
   }
 
   def remove(id: WorkflowId): Boolean = {
-    val newWorkflowStore = workflowStore filterNot { _.id == id }
-
-    if (newWorkflowStore == workflowStore) {
-      false
-    } else {
-      workflowStore = newWorkflowStore
+    if (workflowStore.exists(_.id == id)) {
```

Comment:
Yeah I was more thinking in terms of memory used when the list is duplicated unnecessarily. But we probably already spent too much time writing about this than it's worth xD I'll revert it and it will be modified when the DB comes in anyway like you said.